File: src/main/java/com/example/photoroulette/model/SwipeAction.kt
package com.example.photoroulette.model

enum class SwipeAction(
    val storageValue: String,
) {
    Skip("skip"),
    Delete("delete"),
    Previous("previous"),
    Next("next"),
    ;

    companion object {
        fun fromStorageValue(value: String?): SwipeAction? = entries.firstOrNull {
            it.storageValue == value
        }
    }
}

File: Flutter_Project/lib/features/settings/domain/models/swipe_action.dart
enum SwipeAction {
  skip('skip'),
  delete('delete'),
  previous('previous'),
  next('next');

  const SwipeAction(this.storageValue);

  final String storageValue;

  static SwipeAction? fromStorageValue(String? value) {
    for (final SwipeAction action in SwipeAction.values) {
      if (action.storageValue == value) {
        return action;
      }
    }

    return null;
  }
}
